assets/js/admin/lesson-edit.js

```diff
@@ -33,4 +31,33 @@ domReady( () => {
 	if ( moduleOptionElements.length > 0 ) {
 		moduleOptionElements.select2( { width: 'resolve' } );
 	}
+
+	// Refresh the prerequisite meta box when to course changes in order to get the relevent prerequisites.
+	jQuery( '#lesson-course-options' ).on( 'change', function () {
+		const lessonId = jQuery( '#post_ID' ).val();
+		const courseId = jQuery( this ).val();
+
+		jQuery.get(
+			ajaxurl,
+			{
+				action: 'get_prerequisite_meta_box_content',
+				lesson_id: lessonId,
+				course_id: courseId,
+				security:
+					window.sensei_lesson_metadata
+						.get_prerequisite_meta_box_content_nonce,
+			},
+			function ( response ) {
+				if ( '' != response ) {
+					// Replace the meta box and re-initialize select2.
+					jQuery( '> .inside', '#lesson-prerequisite' ).html(
+						response
+					);
+					jQuery( '#lesson-prerequisite-options' ).select2( {
```

Comment:
Could we use the `prerequisiteOptionElements` constant here?

Reply:
We can't. The element reference is lost after the content is replaced, so we need to target it again.